modules/tensor_mechanics/include/materials/ComputeLagrangianStressSmall.h

```diff
@@ -0,0 +1,79 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ComputeLagrangianStressCauchy.h"
+
+/// Provide the Cauchy stress via an objective integration of a small stress
+//
+//    This class provides the Cauchy stress and associated Jacobian through
+//    an objective integration of the small strain constitutive model.
+//
+//    The small strain model implements the computeQpSmallStress()
+//    which must provide the _small_stress and _small_jacobian
+//    properties.
+//
+//    This class is then responsible for completing the cauchy stress
+//    update with an objective integration, providing _cauchy_stress
+//    and _cauchy_jacobian properties
+class ComputeLagrangianStressSmall : public ComputeLagrangianStressCauchy
+{
+public:
+  static InputParameters validParams();
+  ComputeLagrangianStressSmall(const InputParameters & parameters);
+  virtual ~ComputeLagrangianStressSmall(){};
+
+protected:
+  /// Implement the objective update
+  virtual void computeQpCauchyStress() override;
+
+  /// Method to implement to provide the small stress update
+  //    This method must provide the _small_stress and _small_jacobian
+  virtual void computeQpSmallStress() = 0;
```

Comment:
Maybe rename to `computeQpObjectiveStress`?

Reply:
The user is implementing the small stress update.  So I think this one is named correctly.